iter/chain.go

```diff
@@ -14,6 +14,12 @@ func Chain[T any](iterators ...Iterator[T]) *ChainIter[T] {
 	return &ChainIter[T]{iterators, 0}
 }
 
+// Filter instantiates a [*FilterIter] that selectively yields only results
+// that cause the provided function to return `true`.
```

Comment:
Since this is documented on the Filter iterator, we can just signpost people to it rather than redocument:

```
// Filter is a convenience method for [Filter], providing this iterator as
// as argument.
```

The [Filter] will magically create a hyperlink in the generated docs.

Reply:
Same applies for every `.Filter` method you've added.